src/pybind/mgr/cephadm/serve.py

```diff
@@ -1438,12 +1438,20 @@ async def _get_container_image_info(self, image_name: str) -> ContainerInspectIn
         if self.mgr.cache.host_needs_registry_login(host) and self.mgr.registry_url:
             await self._registry_login(host, json.loads(str(self.mgr.get_store('registry_credentials'))))
 
-        pullargs: List[str] = []
-        if self.mgr.registry_insecure:
-            pullargs.append("--insecure")
-
-        j = await self._run_cephadm_json(host, '', 'pull', pullargs, image=image_name, no_fsid=True)
-
+        j = None
+        try:
+            j = await self._run_cephadm_json(host, '', 'inspect-image', [],
+                                             image=image_name, no_fsid=True)
+        except OrchestratorError:
+            pass
+
+        if not j:
+            pullargs: List[str] = []
+            if self.mgr.registry_insecure:
+                pullargs.append("--insecure")
+
+            j = await self._run_cephadm_json(host, '', 'pull', pullargs,
+                                             image=image_name, no_fsid=True)
```

Comment:
It seems like the second check (the one running `pull`) could go into the except clause.

Reply:
maybe report the exception cause via `log.debug` ?

Reply:
It is rather expected situation that "inspect" may fail, and we have similar code in upgrade [1] and we don't have any debug loggin there. And we have enough debug logging in `_run_cephadm`, which will be called, so one would see the result. Still if you think a `log.debug` message would be useful here I don't mind adding it.

[1] https://github.com/ceph/ceph/blob/main/src/pybind/mgr/cephadm/upgrade.py#L836

Reply:
ah yep! the debug logging in `_run_cephadm` is sufficient, there really is no need for additional logging here.